### PR TITLE
Change usage of library

### PR DESCRIPTION
**Pull request by MrWook**
Closes: https://github.com/zxcvbn-ts/zxcvbn/issues/194

We move away from the painfull singleton usage to a class based approach to handle multiple options at the same time and make it less error prone it different setups.
This way we get the setup like this:
```ts
import { ZxcvbnFactory } from '@zxcvbn-ts/core'
import * as zxcvbnEnPackage from '@zxcvbn-ts/language-en'

const options = {
  translations: zxcvbnEnPackage.translations,
}
const zxcvbn = new ZxcvbnFactory(options)

const password = 'somePassword'
zxcvbn.check(password)
```

### CODE REVIEW

File: packages/libraries/main/src/utils/mergeUserInputDictionary.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    const isWordSizeBigger =
+      rankedDictionariesMaxWordSize.userInputs <
+      userInputsOptions.rankedDictionaryMaxWordSize
+
+    rankedDictionaries.userInputs = {
+      ...rankedDictionaries.userInputs,
+      ...userInputsOptions.rankedDictionary,
+    }
+
+    rankedDictionariesMaxWordSize.userInputs = isWordSizeBigger
```
Comment (camerondm9) [inline packages/libraries/main/src/utils/mergeUserInputDictionary.ts:31]: Couldn't this just be `Math.max(userInputsOptions.rankedDictionaryMaxWordSize, rankedDictionariesMaxWordSize.userInputs)`?

Then you wouldn't need to calculate `isWordSizeBigger` either.

File: packages/libraries/main/test/scoring/guesses/calc.spec.ts [hunk 2]
```diff
@@ -21,15 +23,15 @@ describe('scoring', () => {
...
+    expect(estimate(zxcvbnOptions, match, '1977')).toEqual({
       pattern: 'date',
       token: '1977',
       year: 1977,
       month: 7,
       day: 14,
       // @ts-ignore
       guesses: dateGuesses(match),
-      guessesLog10: 4.225050696138048,
+      guessesLog10: 4.234390722392192,
```
Comment (camerondm9) [inline packages/libraries/main/test/scoring/guesses/calc.spec.ts:34]: Why is this changing?

I would expect the results to be the same, if this PR is just refactoring.

Comment (MrWook) [reply packages/libraries/main/test/scoring/guesses/calc.spec.ts:34]: I would think the same too and it has nothing to do with this changes here.
We got a new year so the date matcher scoring will match the date differently. I guess i really need to change the test date to always be x years behind and not change

File: packages/libraries/main/src/utils/mergeUserInputDictionary.ts (hunk 1 above)
Comment (MrWook) [reply packages/libraries/main/src/utils/mergeUserInputDictionary.ts:31]: Thank you, i always forget that `Math.max` is a thing 👍

**Pull request merged**